contrib/docker/99-init-pixelfed.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+APPKEY=""
+EXCLUDES=""
+
+# Create the storage tree if needed
+cp -r storage.skel/* storage/
+
+# run nginx workers with the 'application' user
+sed -i 's/^user.*;$/user application;/' /etc/nginx/nginx.conf
+
+# Check for APP_KEY in .env_app_key
+if ! grep -q -E "^APP_KEY=['\"]?base64:" /app/storage/.env_app_key 2>/dev/null; then
+  APPKEY="$(php artisan key:generate --show)"
+  echo "APP_KEY=\"${APPKEY}\"" > /app/storage/.env_app_key
+else
+  echo " * skipping generation of APP_KEY, existing key found in .env_app_key"
+fi
+
+# Check for APP_HASHID_SALT in .env_app_key
+if ! grep -q -E "^APP_HASHID_SALT=['\"]?[A-Za-z0-9]+" /app/storage/.env_app_hashid_salt 2>/dev/null; then
+  echo 'APP_HASHID_SALT="'$(tr -dc A-Za-z0-9 </dev/urandom | head -c 129)'"' > /app/storage/.env_app_hashid_salt
+else
+  echo " * skipping generation of APP_HASHID_SALT, existing salt found in .env_app_hashid_salt"
+fi
+
+# check for TRUST_PROXIES and configure nginx real_ip module
+TRUST_PROXIES="${TRUST_PROXIES:=false}"
+if [[ "$TRUST_PROXIES" == "false" ]]; then
+  echo "set_real_ip_from 0.0.0.0/0;" >> /opt/docker/etc/nginx/vhost.common.d/10-realip.conf
+else
+  for ip in ${TRUST_PROXIES//,/ }; do
+    echo "set_real_ip_from ${ip};" >> /opt/docker/etc/nginx/vhost.common.d/10-realip.conf
+  done
+fi
+
+# sanitize env variables and transfer to temporary .env file
+EXCLUDES_DEF="APPLICATION|CONTAINER|DOCKER|GPG_KEYS|HOME|HOSTNAME|LANG|LC_|LOG_|NSS|OLDPWD|PATH|PHP|PWD|SERVICE_|SHLVL|TERM|WEB|_"
+EXCLUDES_K8S="KUBERNETES|OPENSHIFT|_PORT|_SERVICE"
+env | grep -Ev "^(${EXCLUDES_DEF})|${EXCLUDES_K8S}" > /app/storage/.env_temp
+
+# combine APP_KEY, APP_HASHID_SALT and ENVs int .env file
+cat /app/storage/.env_app_key \
+    /app/storage/.env_app_hashid_salt \
+    /app/storage/.env_temp \
+    | sort > /app/storage/.env
+
+# Create actors if AP is enabled
+ACTIVITY_PUB="${ACTIVITY_PUB:=false}"
+if [[ ${ACTIVITY_PUB} == "true" ]]; then
+  php artisan instance:actor
+fi
+
+# Create OAuth keys if enabled
+OAUTH_ENABLED="${OAUTH_ENABLED:=false}"
```

Comment:
This can also be set by PASSPORT_PRIVATE_KEY. See https://github.com/pixelfed/pixelfed/pull/4257

Reply:
Understood, I'll look into how to include this as well.